syntax = "proto3";
import "google/protobuf/empty.proto";

// to compile
// python3 -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. --pyi_out=. ./insultingServer.proto

message Insult {
  string insult = 1;
  float severity = 2;
}

message Insults {
  repeated string value = 1;
}

message InsultName {
  string insult_name = 1;
}

message Severity {
  float value = 1;
}

service InsultingService {
  rpc GetInsults (google.protobuf.Empty) returns (Insults) {}
  rpc AddInsult (Insult) returns (google.protobuf.Empty) {}
  rpc InsultMe (google.protobuf.Empty) returns (InsultName) {}
  rpc GetSeverity(InsultName) returns (Severity) {}
}
